pkg/serverless/daemon/daemon.go

```diff
@@ -337,13 +340,17 @@ func (d *Daemon) Stop() {
 func (d *Daemon) TellDaemonRuntimeStarted() {
 	// Reset the RuntimeWg on every new invocation.
 	// We might receive a new invocation before we learn that the previous invocation has finished.
+	d.runtimeStateMutex.Lock()
+	defer d.runtimeStateMutex.Unlock()
 	d.RuntimeWg = &sync.WaitGroup{}
 	d.TellDaemonRuntimeDoneOnce = &sync.Once{}
 	d.RuntimeWg.Add(1)
```

Comment:
do you want to move this one line up as well?

Reply:
I don't think it makes a difference with the mutex